### ISSUE TITLE
Bug in 'amy' langauge

### ISSUE THREAD

**Issue by linas** [opencog/link-grammar#421]
This is inspired by opencog/relex/issues/248 -- for the 'amy' langauge, I get:
```
linkparser> 'relex' is the best
link-grammar: Warning: Internal error: word "'relex'" got issued more than once

link-grammar: Warning: Internal error: word "relex'" got issued more than once

separate_word: Contracted word part ' is not in the dict
separate_word: Contracted word part ' is not in the dict
Warning: Invalid morpheme type combination 'wwtwwtststs'.
Run with !bad and !verbosity>4 to debug
Warning: Invalid morpheme type combination 'wwtwwwtsts'.
Run with !bad and !verbosity>4 to debug
Warning: Invalid morpheme type combination 'wwtwwtsww'.
Run with !bad and !verbosity>4 to debug
Warning: Invalid morpheme type combination 'wtwwtsww'.
Run with !bad and !verbosity>4 to debug
Warning: Invalid morpheme type combination 'wwtwwtstsw'.
Run with !bad and !verbosity>4 to debug

```

**Comment by linas** [comment]
Never mind. Just fixed it

**Comment by linas** [comment]
fixed in #422

**Comment by ampli** [comment]
Just a note: This fixes this particular problem.
The internal error is because my "wordgraph flattening" algo is an ad-hoc one, that may still have such a problem in other situations that we still didn't encounter. However, in order to discover such cases it makes some consistency checks, among them validating that it doesn't issue a token more than once.
